server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchUsersAction.java

```diff
@@ -71,6 +99,29 @@ public void define(WebService.NewController context) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      OrganizationDto organization = wsSupport.getOrganizationByKey(dbSession, request.mandatoryParam(PARAM_ORGANIZATION));
+      QProfileDto profile = wsSupport.getProfile(dbSession, organization, request.mandatoryParam(PARAM_QUALITY_PROFILE), request.mandatoryParam(PARAM_LANGUAGE));
+      wsSupport.checkCanEdit(dbSession, profile);
+      List<QProfileEditUserMembershipDto> result = searchResult(dbSession, request, organization, profile);
+    }
     IOUtils.write(IOUtils.toString(getClass().getResource("search_users-example.json")), response.stream().output());
   }
+
+  private List<QProfileEditUserMembershipDto> searchResult(DbSession dbSession, Request request, OrganizationDto organization, QProfileDto profile){
+    return dbClient.qProfileEditUsersDao().selectByQuery(dbSession,
+      builder()
+        .setOrganization(organization)
+        .setProfile(profile)
+        .setQuery(request.param(TEXT_QUERY))
+        .setMembership(MEMBERSHIP.get(fromParam(request.mandatoryParam(SELECTED))))
+        .build(),
+      forPage(request.mandatoryParamAsInt(PAGE))
+        .andSize(request.mandatoryParamAsInt(PAGE_SIZE)));
+  }
+
+  private void writeResponse(List<QProfileEditUserMembershipDto> result){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1186)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "writeResponse" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make "writeResponse" a "static" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "result". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)